sale_commission_settlement_payment_date/wizard/wizard_settle.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2023 Nextev
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import json
+from datetime import datetime
+
+from odoo import fields, models
+
+
+class SaleCommissionMakeSettle(models.TransientModel):
+    _inherit = "sale.commission.make.settle"
+
+    date_payment_to = fields.Date("Date Payment Up to", default=fields.Date.today())
+
+    def _get_agent_lines(self, agent, date_to_agent):
+        """
+        Exclude invoices with invoice payment date is after date_payment_to
+        """
+        agent_lines = super()._get_agent_lines(agent, date_to_agent)
+        for line in agent_lines.filtered(
+            lambda r: r.commission_id.invoice_state == "paid"
+            and self.date_payment_to != "false"
+        ):
```

Comment:
`date_payment_to` will always be different from the string `"false"` since it can only be a Date object or the boolean `False`

In general, I don't think this particular check should go in the lambda since it's not meant to filter `agent_lines` but it's meant to check if the filtering should happen at all. Additionally I'd group the other checks together since they're not logically separate.

```python
if date_payment_to:
    for line in agent_lines.filtered(lambda r: r.commission_id.invoice_state == "paid" and r.invoice_id.invoice_payments_widget and not self.check_payment_date(r)):
```

or 

```python
if date_payment_to:
    for line in agent_lines:
        if line.commission_id.invoice_state == "paid" and line.invoice_id.invoice_payments_widget and not self.check_payment_date(line):
```

(and then let Black format that of course :) )

Reply:
Here condition on "false" it's my mistake. I will change it and re-publish all changes